bentoml/_internal/frameworks/mlflow.py

```diff
@@ -0,0 +1,213 @@
+            example: ``{"training-set": "data-v1"}``.
+        custom_objects:
+            Custom objects to be saved with the model. An example is
+            ``{"my-normalizer": normalizer}``. Custom objects are serialized with
+            cloudpickle.
+        metadata:
+            Metadata to be associated with the model. An example is ``{"param_a": .2}``.
+
+            Metadata is intended for display in a model management UI and therefore all
+            values in metadata dictionary must be a primitive Python type, such as
+            ``str`` or ``int``.
+
+    Returns:
+        A :obj:`~bentoml.Model` instance referencing a saved model in the local BentoML
+        model store.
+
+    Example:
+
+    .. code-block:: python
+
+        import bentoml
+
+        bentoml.mlflow.import_model(
+            'my_mlflow_model',
+            model_uri="runs:/<mlflow_run_id>/run-relative/path/to/model",
+            signatures={
+                "batchable": True
+            }
+        )
+    """
+    context = ModelContext(
+        framework_name="mlflow",
+        framework_versions={"mlflow": mlflow.__version__},
+    )
+
+    if signatures is None:
+        signatures = {
+            "predict": {"batchable": False},
+        }
+        logger.info(
+            f"Using the default model signature for MLFlow pyfunc model ({signatures}) for model {name}."
+        )
+    if len(signatures.keys()) != 1 or "predict" not in signatures:
+        raise BentoMLException(
+            f"MLFlow pyfunc model support only the `predict` method, signatures={signatures} is not supported"
+        )
+
+    with bentoml.models.create(
+        name,
+        module=MODULE_NAME,
+        api_version=API_VERSION,
+        signatures=signatures,
+        labels=labels,
+        options=None,
+        custom_objects=custom_objects,
+        metadata=metadata,
+        context=context,
+    ) as bento_model:
+        from mlflow.models import Model as MLflowModel
+        from mlflow.models.model import MLMODEL_FILE_NAME
+        from mlflow.tracking.artifact_utils import _download_artifact_from_uri
+
+        local_path = _download_artifact_from_uri(
+            artifact_uri=model_uri, output_path=bento_model.path_of(MLFLOW_MODEL_FOLDER)
+        )
+
+        mlflow_model_file = os.path.join(local_path, MLMODEL_FILE_NAME)
+        if not os.path.exists(mlflow_model_file):
+            raise BentoMLException("artifact is not a mlflow model")
+
+        model_meta = MLflowModel.load(mlflow_model_file)
+        conf = model_meta.flavors.get("pyfunc")
+        if conf is None:
+            raise BentoMLException(
+                "Target MLFlow model does not support python_function flavor"
+            )
+
+        return bento_model
+
+
+def get_runnable(bento_model: bentoml.Model) -> t.Type[bentoml.Runnable]:
+    """
+    Private API: use :obj:`~bentoml.Model.to_runnable` instead.
+    """
+    assert bento_model.info.signatures.keys() == ["predict"]
```

Comment:
we shouldn't use assert in most cases. a little optimization. Maybe a raise exception here makes more sense

Reply:
this is actually a good case to use assert instead of raising exception, it's an internal assumption and serves as documentation for ppl reading the code here